test/plugin/download_endpoint_test.rb

```diff
@@ -138,6 +170,34 @@ def endpoint
     url2 = @uploaded_file.url
     assert_equal url1, url2
   end
+  it "returns same download_url regardless of metadata order" do
+    @uploaded_file.data["metadata"] = { "filename" => "a", "mime_type" => "b", "size" => "c" }
+    url1 = @uploaded_file.download_url
+    @uploaded_file.data["metadata"] = { "mime_type" => "b", "size" => "c", "filename" => "a" }
+    url2 = @uploaded_file.download_url
+    assert_equal url1, url2
+  end
+
+  it "returns signature and expires_at when configured" do
+    secret = SecureRandom.hex(64)
```

Comment:
I'd almost prefer a static secret here.

1. We don't have to generate it from entropy.
2. During test failures it's not going to cause issues recreating them.

Maybe something like `"A" * 64` assigned to a constant near the top of the test. WDYT?